src/EFCore.PG/Migrations/NpgsqlMigrationsSqlGenerator.cs

```diff
@@ -519,7 +519,8 @@ protected override void Generate(RenameTableOperation operation, [CanBeNull] IMo
                 name = operation.NewName;
             }
 
-            if (operation.NewSchema != null)
+            if (operation.NewSchema != null &&
+                operation.NewSchema != operation.Schema)
```

Comment:
In #702 I found that other rename operations could move objects to a new schema. So we need the same handling for indexes, sequences, etc. And let's add a method which will check that the operation schema name was changed.